### PR TITLE
Add backup support for all versions

### PR DESCRIPTION
**Pull request by swjain23**
- support PG version 9.5, 9.6, 10, 11, 12
- Added executables and dlls for all the versions
- Gets the server version and matches the folder to figure out which exe to use.

### CODE REVIEW

File: ossdbtoolsservice/disaster_recovery/disaster_recovery_service.py [hunk 1]
```diff
@@ -143,27 +150,50 @@ def _get_backup_restore_connection_params(connection_options: dict) -> List[str]
...
+    for folder in all_folders:
+        folderName = os.path.basename(folder)
+        version = folderName.split('.')
+        # Get the major version value
+        try:
+            major = int(version[0])
+        except ValueError:
+            major = 0
+        minor = 0
+        # Set minor version if version length is more than 1 (ex 9.5, 9.6)
```
Comment (rgreenMSFT) [inline ossdbtoolsservice/disaster_recovery/disaster_recovery_service.py:186]: Wouldn't minor version be version[1]?  What if MySQL releases 10 (not 10.0)?  Would this code still work?

File: ossdbtoolsservice/disaster_recovery/disaster_recovery_service.py [hunk 2]
```diff
@@ -143,27 +150,50 @@ def _get_backup_restore_connection_params(connection_options: dict) -> List[str]
...
+        if (len(version) > 1):
+            minor = int(version[1])
```
Comment (rgreenMSFT) [inline ossdbtoolsservice/disaster_recovery/disaster_recovery_service.py:188]: Do you need try..except for ValueError here as well?

File: ossdbtoolsservice/disaster_recovery/disaster_recovery_service.py [hunk 3]
```diff
@@ -143,27 +150,50 @@ def _get_backup_restore_connection_params(connection_options: dict) -> List[str]
...
+        path_suffix = os.path.join('bin', exe_name)
+
+    # Get the list of folders in the os specific root folder
+    all_folders: List[str] = [os.path.normpath(x[0]) for x in os.walk(os_root)]
+    for folder in all_folders:
+        folderName = os.path.basename(folder)
+        version = folderName.split('.')
+        # Get the major version value
+        try:
+            major = int(version[0])
```
Comment (rgreenMSFT) [inline ossdbtoolsservice/disaster_recovery/disaster_recovery_service.py:182]: Shouldn't we do this assignment outside the for loop?  No need to run the conversion on each iteration, I think.

File: ossdbtoolsservice/disaster_recovery/disaster_recovery_service.py [hunk 4]
```diff
@@ -143,27 +150,50 @@ def _get_backup_restore_connection_params(connection_options: dict) -> List[str]
...
+        if (len(version) > 1):
+            minor = int(version[1])
+
+        if major == int(server_version[0]) and minor == server_version[1]:
+            exe_path = os.path.join(folder, path_suffix)
+            if not os.path.exists(exe_path):
+                raise ValueError(f'Could not find executable file {exe_path}')
+            return exe_path
+
+    raise ValueError(f'Exe folder {os_root} does not contain {exe_name}')
```
Comment (rgreenMSFT) [inline ossdbtoolsservice/disaster_recovery/disaster_recovery_service.py:196]: This doesn't have the file name extension, does it?  Shouldn't we include that in the error message?

Comment (rgreenMSFT) [reply ossdbtoolsservice/disaster_recovery/disaster_recovery_service.py:196]: Should we append path_suffix instead?

Comment (swjain23) [reply ossdbtoolsservice/disaster_recovery/disaster_recovery_service.py:196]: Yes it would not contain the extension, will use suffix instead

**Pull request merged**